src/Simulations/simulation.jl

```diff
@@ -80,7 +84,22 @@ function set!(
         # we need to use the reinit function because we don't 
         # have direct state access using DiffEqBase. For this
         # we need to copy
-        Y = copy(simulation.integrator.u)
+        if simulation.restart isa NoRestart
+            Y = copy(simulation.integrator.u)
+            tf = simulation.integrator.sol.prob.tspan[2]
```

Comment:
I would give `tf` a more explicit varname, e.g., `time_end`

Reply:
> Just to clarify, does this only add the restart option, or does it also add the ability for saving at user-defined time intervals (as in the description)?
> 
> Edit: I see there is a change in the JLD2Output that allows output at user-defined time intervals.

1) It allows outputs (JLD2) at user defined time intervals. 
2) It allows restarts using a user-specified (JLD2) checkpoint file (which is the same format as the "output" file), since they store solution objects (in these instances, an `integrator` that can be used to define `ClimaCore Fields)

Reply:
1) It allows outputs (JLD2) at user defined time intervals. 
2) It allows restarts using a user-specified (JLD2) checkpoint file (which is the same format as the "output" file), since they store solution objects (in these instances, an `integrator` that can be used to define `ClimaCore Fields)

Reply:
@jiahe23 resolved tf to `time_final`